bbinc/fwd_types.h

```diff
@@ -0,0 +1,35 @@
+/*
+   Copyright 2015 Bloomberg Finance L.P.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+/*
+ * forward declared types
+ */
+
+#ifndef __fwd_types_h__
+#define __fwd_types_h__
+
+typedef struct bdb_state_tag bdb_state_type;
+typedef struct UnpackedRecord UnpackedRecord;
+typedef struct Vdbe Vdbe;
```

Comment:
It feels wrong to place SQLite definitions in `bbinc` and have SQLite include the file.

Reply:
Yeah, maybe i'll move it to sqlite/